### ISSUE TITLE
Errors on caRpools execution

### ISSUE THREAD

**Issue by AGV69** [boutroslab/caRpools#2]
Dear Jan,

When I execute caRpools (ver. 0.83) on RStudio using the demo datasets (_CRISPR_KO3_Drug, Wang or Shalem_), I got the following errors (sometimes error [1] and other error [2]):

[1] Quitting from lines 696-708 (CaRpools-extended-PDF.Rmd) 
Error in wilcox.test.default(x$foldchange, control.test, alternative = "t") : 
  not enough 'y' observations

[2] Quitting from lines 765-770 (CaRpools-extended-PDF.Rmd) 
Error in `rownames<-`(`*tmp*`, value = c("CTRL", "CTRL", "TREAT", "TREAT" : 
  duplicate rownames not allowed

I am not able to complete the script running. Could you give me a solution?
Thanks in advance.

Alberto

**Comment by jwinter6** [comment]
Hi Alberto,

Sorry for the late reply.
I will have a look into it in the next days.
Right now we currently work on a fully interactive version, please let me know if you are interested in testing the beta version, which I would highly suggest.

Best
Jan

**Comment by AGV69** [comment]
Hi Jan,

Thanks for you reply. I am very interested to test the new beta version.
Can you explain me the way to do it?

Regards.

Alberto


2016-07-18 23:38 GMT+02:00 Jan Winter <notifications@github.com>:

**Comment by AliaksandraR** [comment]
Hi,
 
I get the same error:
Quitting from lines 696-708 (CaRpools-extended-PDF.Rmd) 
Error in wilcox.test.default(x$foldchange, control.test, alternative = "t") : 
not enough 'y' observations

Could you please suggest a solution?
Best,
Alex

**Comment by AliaksandraR** [comment]
Also, got the second one, mentioned in the first post by Jan. Maybe there is a solution available already?
Thanks in advance,
Alex

**Comment by jwinter6** [comment]
Hi Alex,

I am still working on a newer version of it (many things have changed, I have uploaded an alpha version couple of weeks ago) which will also have a fix for that.
In the meantime I could offer you a test spot of a beta version of caRpoolsInteractive if you like (which might convince you even more :) )

cheers
Jan

**Comment by AliaksandraR** [comment]
Hi Jan,

It would be great to test the interactive version. How do I get it?
I will also give a try to the newer version.
Best,
Alex

**Comment by AGV69** [comment]
Hi Jan,

I am also interested to test the interactive version. Please, how do I get
it?

The newer alpha version is available on GitHub? Is it the Alpha release
1.025?

Best.

Alberto

2016-08-10 14:24 GMT+02:00 AliaksandraR <notifications@github.com>:

> Hi Jan,
>
> It would be great to test the interactive version. How do I get it?
> I will also give a try to the newer version.
> Best,
> Alex
>
>

**Comment by jwinter6** [comment]
Could you please send me your email addresses to jan.winter@dkfz-heidelberg.de ?
I will send out invites to you then :)
cheers
Jan

**Comment by AGV69** [comment]
I did it! Thanks,

Alberto

2016-08-17 12:20 GMT+02:00 Jan Winter <notifications@github.com>:

**Comment by AliaksandraR** [comment]
Hi Jan,

At the moment I am using the latest ALPHA 1.025 R package and have two errors. 

The first one upon the execution of the stat.DESeq function and it is:

**Error in `rownames<-`(`*tmp*`, value = c("CTRL", "CTRL", "TREAT", "TREAT" : 
  duplicate rownames not allowed**

The second one upon the execution of the stat.mageck function and it is:

**usage: mageck [-h] [-v] {count,test,pathway,plot,mle} ...
mageck: error: unrecognized arguments: Drive/BRIC/R analysis/caRpools/mageckanalysisfile_MAGeCK_sgRNA.tab
Error in file(file, "rt") : cannot open the connection
In addition: Warning message:
In file(file, "rt") :
  cannot open file 'mageckanalysisfile.gene_summary.txt': No such file or directory**

Could you please provide some help with those?

Thanks,
Alex

**Comment by AGV69** [comment]
Hi Jan,

Thanks for the access to the online versión of caRpools. It's a nice tool.
Congratulation!

I tried caR with a readcounts file and all is OK but I can't download the
"Hit calling plots" file. I got the following error:

The proxy server received an invalid response from an upstream server.
The proxy server could not handle the request GET
/session/6e7372237d722b7533e23ff53c468995/download/downloads_hcPlots.

Reason: Error reading from remote server

I can download the other files (Screen quality and Hit calling data). Can
you help me? Thanks in advance.

Alberto

2016-08-24 15:29 GMT+02:00 AliaksandraR <notifications@github.com>:

**Comment by jwinter6** [comment]
Hi Alberto,

glad you like it :)
We will provide a huge update soon (final update), which resolves many issues including yours.

cheers
Jan

**Comment by reilya** [comment]
Dear Jan,

I have installed the alpha release on a Mac and on a Linux box. The installation went well, but when it comes to running the example I get the following error when running the example

Quitting from lines 31-40 (CaRpools-extended-PDF-extended_ALPHA.Rmd) 
Error in openxlsx::read.xlsx(miaccs.file, sheet = "MIACCS", startRow = 1,  : 
  object 'miaccs.file' not found

This is using test=FALSE. With test=true I get

Error in load.file(paste(getwd(), "/", miaccs, sep = ""), type = "xlsx") : 
  No library FASTA file provided. Please provide the filename including the .fasta extension of your library.

It seems the program is not finding the miaccs file. I checked the paths and everything seems to be ok.

Thanks in advance
Alex

**Comment by jwinter6** [comment]
Hi,

Since I am in the last week's of work for the interactive caRpools (will be available soon),  I can just guess it is a mac thing I observed before.
Have you tried to open rstudio via command line?

Try

open -a rstudio

And let me know whether it helped.
If not, I will have a look into it.

Cheers
Jan

Am 16. September 2016 16:56:20 schrieb reilya <notifications@github.com>:

Dear Jan,

I have installed the alpha release on a Mac and on a Linux box. The installation went well, but when it comes to running the example I get the following error when running the example

Quitting from lines 31-40 (CaRpools-extended-PDF-extended_ALPHA.Rmd)
Error in openxlsx::read.xlsx(miaccs.file, sheet = "MIACCS", startRow = 1, :
object 'miaccs.file' not found

This is using test=FALSE. With test=true I get

Error in load.file(paste(getwd(), "/", miaccs, sep = ""), type = "xlsx") :
No library FASTA file provided. Please provide the filename including the .fasta extension of your library.

It seems the program is not finding the miaccs file. I checked the paths and everything seems to be ok.

Thanks in advance
Alex

-

**Comment by OekoGuesel** [comment]
I also have the following error (using caRpools v0.83):

Quitting from lines 734-739 (CaRpools-PDF.Rmd)
Error in `rownames<-`(`*tmp*`, value = c("CTRL", "CTRL", "TREAT", "TREAT" :
  duplicate rownames not allowed

Is there no fix or reliable workaround yet?

**Comment by jwinter6** [comment]
Hi OekoGuesel,

I had a look trying to understand the issue posted with the row names and was not able to reproduce the error.
Can you make sure the following is true:
- no duplicate entries in the sgRNA FASTA library file
- The same file is not assigned to two groups at the same time
- In case you use read count files -> no duplicate entries there as well

In the meantime, please write me an email to jan.winter@dkfz-heidelberg.de as I could give you access to a beta version of our new analysis software.
Nevertheless I will have another look, as it also occurred in the pre-release alpha version.

**Comment by skalava23** [comment]
Hi Jan,

I'm getting the same error 

Quitting from lines 734-739 (CaRpools-PDF.Rmd)
Error in rownames<-(*tmp*, value = c("CTRL", "CTRL", "TREAT", "TREAT" :
duplicate rownames not allowed

Has this bug been resolved yet? I checked all of the data files and there are no duplicates in the rownames. I would rather use caRpools on R than on the analysis software, and I would really appreciate if you could help me resolve this bug.

Thanks in advance!

**Comment by dchiu911** [comment]
Hi @jwinter6, 

I have also run into the rownames error reported by some here. It occurs because we are trying to assign duplicated column names from `countdata` to the row names of `coldata` (See in L24 of `DESeq2::DESeqDataSetFromMatrix`)

In `caRpools::stat.DESeq`, the `countdata` column names are currently assigned as such:
```r
colnames(countdata) = c(rep("CTRL", length(untreated.list)),
                        rep("TREAT", length(treated.list)))
colnames(countdata)
#> [1] "CTRL"  "CTRL"  "TREAT" "TREAT"
```
One possible fix can be
```r
colnames(countdata) = c(paste0("CTRL", seq_along(untreated.list)),
                        paste0("TREAT", seq_along(treated.list)))
colnames(countdata)
#> [1] "CTRL1"  "CTRL2"  "TREAT1" "TREAT2"
```
Are you able to release a new version of `caRpools` with this bug resolved?
Thanks!

**Comment by jwinter6** [comment]
Hi and many thanks for looking into this.
unfortunately I cannot release a new version in the next 2 months as I am busy with my thesis.
I just wonder why in the alpha this workaround is not implemented, I was so sure I added this ages ago :(

In the meantime, please go on with CRISPRAnalyzeR, which offers way more :)

Anyway, as soon as thesis stuff is done I will upload a hotfix on that.
Thanks for your help
Jan

**Comment by yeroslaviz** [comment]
Hi,
the solution from @dchiu911 works nicely. 
I would suggest to add it the the package. 

Especially because the CripsrAnalyzeR does not work for me. and it doesn't seems to be able to work without the interactive part.

**Comment by jwinter6** [comment]
Hi Frymor,

CRISPRAnalyzer is a Shiny application and thus not able to run without the interactive GUI - this is the main point of CRISPRAnalyzeR which is aimed at non-bioinformaticians.
If it does not work for you, please do me the favour and raise an issue so we can solve why it does not work for you.

The fix will be added once the thesis is done, though new developments (feature-wise) will only be adapted to CRISPRAnalyzeR.

Best
Jan

**Comment by yeroslaviz** [comment]
Hi Jan and thanks for the fast response.
[I have described my problem with the CrisprAnalyzeR  already](https://github.com/boutroslab/CRISPRAnalyzeR/issues/42).

Best
Assa

**Comment by jwinter6** [comment]
Oh sorry, didn't get a notification about a new issue - my fault.
I will look into this the next days, I saw that Kitematic is causing a lot of trouble on macOS, and I experienced many of them as well.

**Comment by yeroslaviz** [comment]
no worries. I am still working out all the kinks from the caRpools scripts, which are already awesome by themselves. But it would be good, if in the new version of the R package you can modify the column names in the header of the count table, both of the sgRNA and the genes objects.

**Comment by ktretina** [comment]
I'm running into this same problem, and would like to use this package as soon as possible. Is this going to be fixed some time soon? Thanks!
